src/test/kotlin/com/example/flashcardbackend/deck/DeckControllerTest.kt

```diff
@@ -0,0 +1,186 @@
+package com.example.flashcardbackend.deck
+
+import com.example.flashcardbackend.flashcard.FlashcardListItem
+import org.hamcrest.Matchers
+import org.junit.jupiter.api.Test
+import org.mockito.Mockito.*
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
+import org.springframework.boot.test.mock.mockito.MockBean
+import org.springframework.http.MediaType
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers.*
+
+@WebMvcTest(DeckController::class, DeckService::class)
+class DeckControllerTest(
+    @Autowired
+    private val mockMvc: MockMvc,
+) {
+
+    @MockBean
+    private lateinit var deckRepository: DeckRepository
+
+    @Test
+    fun `should return all decks`() {
+        val deckListItems = listOf(
+            DeckListItem(12, 1, "Deck 1", "Description 1"),
+            DeckListItem(13, 1, "Deck 2", "Description 2"),
+        )
+
+        `when`(deckRepository.findAll()).thenReturn(deckListItems)
+
+        mockMvc.perform(get("/decks"))
+            .andExpect(status().isOk)
```

Comment:
I would split this test into these test methods:

- Should return the HTTP status code OK
- Should return the found decks as JSON.
- Should return two decks.
- Should return the information of the found decks (use `jsonPath()`)

If you split this one test into multiple tests, you maximize the amount of information you get when a test fails. For example, if the system under test would return HTTP status 201, you would have to run your test multiple times before you figure out that the only thing that's wrong is the wrong status code. If you have multiple tests, you have to run your tests only once (well, twice to make sure that the fix is working). Also, the same applies for your other tests.

Reply:
You are right, when something is wrong I have to "run your test multiple times before you figure out that the only thing that's wrong". And this is what I've been doing. :D Your suggestion is better.